Mathlib/CategoryTheory/Distributive/Monoidal.lean

```diff
@@ -0,0 +1,321 @@
+    (coprodComparison (tensorRight X) Y Z) ≫ (β_ (Y ⨿ Z) X).hom =
+    (coprod.map (β_ Y X).hom (β_ Z X).hom) ≫ (coprodComparison (tensorLeft X) Y Z) := by
+  simp [coprodComparison]
+
+/-- A left distributive symmetric monoidal category is right distributive. -/
+instance SymmetricCategory.isMonoidalRightDistrib_of_isMonoidalLeftDistrib
+    [SymmetricCategory C] [IsMonoidalLeftDistrib C] : IsMonoidalRightDistrib C where
+  preservesBinaryCoproducts_tensorRight X :=
+    preservesColimitsOfShape_of_natIso (SymmetricCategory.tensorLeftIsoTensorRight X)
+
+/-- A left distributive symmetric monoidal category is distributive. -/
+instance SymmetricCategory.isMonoidalDistrib_of_isMonoidalLeftDistrib
+    [SymmetricCategory C] [IsMonoidalLeftDistrib C] : IsMonoidalDistrib C where
+
+/-- The right distributivity isomorphism of the a left distributive symmetric monoidal category
+is given by `(β_ (Y ⨿ Z) X).hom ≫ (∂L X Y Z).inv ≫ (coprod.map (β_ X Y).hom (β_ X Z).hom)`. -/
+@[simp]
+lemma SymmetricCategory.rightDistrib_of_leftDistrib
+    [SymmetricCategory C] [IsMonoidalLeftDistrib C] {X Y Z : C} :
+    ∂R X Y Z = (coprod.mapIso (β_ Y X) (β_ Z X)) ≪≫ (∂L X Y Z) ≪≫ (β_ X (Y ⨿ Z)) := by
+  ext
+  · simp
+  · simp
+
+/-- A closed monoidal category is left distributive. -/
+instance MonoidalClosed.isMonoidalLeftDistrib [MonoidalClosed C] :
+    IsMonoidalLeftDistrib C where
+  preservesBinaryCoproducts_tensorLeft X := {
+    preservesColimit := ((ihom.adjunction X).leftAdjoint_preservesColimits).1.1
+  }
+
+/-- The inverse of distributivity isomorphism from the closed monoidal strurcture -/
+@[simp]
+lemma MonoidalClosed.leftDistrib_inv [MonoidalClosed C] {X Y Z : C} :
+    (leftDistrib X Y Z).inv =
+    MonoidalClosed.uncurry
+      (coprod.desc (MonoidalClosed.curry coprod.inl) (MonoidalClosed.curry coprod.inr)) := by
+  simp [← (MonoidalClosed.curry_eq_iff)]
+  have : curry (∂L X Y Z).inv = coprod.desc
+      (coprod.inl ≫ curry (∂L X Y Z).inv) (coprod.inr ≫ curry (∂L X Y Z).inv) := by
+    aesop
+  convert this
+  · rw [← MonoidalClosed.curry_natural_left, whisker_inl_inv_leftDistrib]
+  · rw [← MonoidalClosed.curry_natural_left, whisker_inr_inv_leftDistrib]
+
+attribute [local instance] endofunctorMonoidalCategory
+
+instance isIso_coprodComparison_tensorLeft_of_endofunctors {X Y Z : C ⥤ C} :
+    IsIso (coprodComparison (tensorLeft X) Y Z) :=
+  by
+    refine ⟨?_, ?_, ?_⟩
+    · exact {
+    app (c : C) :=
+    coprodObjIso Y Z (X.obj c) ≪≫ (coprodObjIso (X ⊗ Y) (X ⊗ Z) c).symm |>.hom
+    }
+    · ext c <;> simp [coprodComparison, coprodComparison, coprodObjIso]
+    · ext c
+      simp only [coprodComparison, coprodComparison, coprodObjIso, leftDistrib]
+      aesop
+
+/-- The monoidal structure on the category of endofunctors is left distributive. -/
+instance isMonoidalLeftDistrib_of_endofunctors : IsMonoidalLeftDistrib (C ⥤ C) :=
+  IsMonoidalLeftDistrib.of_isIso_coprodComparisonTensorLeft
```

Comment:
If you import `Mathlib.CategoryTheory.Limits.Preserves.FunctorCategory`, the previous lemma (and the file `FunctorCategoryShapes/BinaryProducts`) becomes unnecessary:
```suggestion
instance isMonoidalLeftDistrib_of_endofunctors : IsMonoidalLeftDistrib (C ⥤ C) where
  preservesBinaryCoproducts_tensorLeft F :=
    inferInstanceAs (PreservesColimitsOfShape _ ((whiskeringLeft C C C).obj F))
```